central/deployment/datastore/datastore_impl_postgres_test.go

```diff
@@ -56,35 +52,19 @@ func (s *DeploymentPostgresDataStoreTestSuite) SetupSuite() {
 
 	s.ctx = context.Background()
 
-	source := pgtest.GetConnectionString(s.T())
-	config, err := postgres.ParseConfig(source)
-	s.Require().NoError(err)
-
-	pool, err := postgres.New(s.ctx, config)
-	s.NoError(err)
-	s.db = pool
-
-	imagePostgres.Destroy(s.ctx, s.db)
-	deploymentPostgres.Destroy(s.ctx, s.db)
-
-	s.gormDB = pgtest.OpenGormDB(s.T(), source)
-	ds := imageDataStore.NewWithPostgres(imagePostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB, false), imagePostgres.NewIndexer(s.db), s.riskDataStore, ranking.ImageRanker(), ranking.ComponentRanker())
-	s.imageDatastore = ds
+	s.testDB = pgtest.ForT(s.T())
 
-	s.mockCtrl = gomock.NewController(s.T())
-	s.riskDataStore = riskMocks.NewMockDataStore(s.mockCtrl)
+	imageDS, err := imageDataStore.GetTestPostgresDataStore(s.T(), s.testDB.DB)
+	s.Require().NoError(err)
+	s.imageDatastore = imageDS
 
-	s.deploymentDatastore, err = newDataStore(
-		deploymentPostgres.NewFullTestStore(s.T(), deploymentPostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)),
-		nil, s.db, nil, nil, s.imageDatastore, nil, nil, s.riskDataStore,
-		nil, nil, ranking.ClusterRanker(), ranking.NamespaceRanker(), ranking.DeploymentRanker())
+	deploymentDS, err := GetTestPostgresDataStore(s.T(), s.testDB.DB)
```

Comment:
This would create a new imageDataStore, different from the one created above by `imageDataStore.GetTestPostgresDataStore(s.T(), s.testDB.DB)` . Does that have any consequences to the test?

Reply:
It creates new instance of the datastore but both instances of the datastore point to the same Postgres table. This is because they share the same DB config.

The unit test focuses on database search not the internal state of the datastore which the image datastore does not have anyways.

Reply:
Makes sense, thanks for clarification.